scripts/example.ts

```diff
@@ -1,43 +1,32 @@
+import { expect } from "chai";
 import { join } from "path";
 
 import firebase = require("../src");
```

Comment:
Is there any concern that this is less "realistic" than doing an `npm install ../` and then doing `require('firebase-tools')`?

Reply:
I don't think this is unreasonable. All our unit tests run via the typescript code. All this is really doing (along with the mocha ts-node register stuff) is compiling on-the-fly so we don't have to do `npm run build` and wait for that, then install or do `require('..')` (which would read the `package.json` and import from `lib/index.js`).

I'm not entirely sure that I'll want to keep this file here (and it's a horrible name too), so I might tackle that sooner rather than later?

I'm trying to think of a good directory to include tests like this and the longer-running Functional tests of the emulator... thoughts?

Reply:
Definitely a bad file name.  I am fine with the `../src` include just asking if it's worth the effort.

I do agree none of these belong in the `scripts` directory.  I'd prefer a `integration-tests` directory where each one is a folder and can ideally be executed with the same invocation so you just do `cd ./$testname && ./run.sh`

Reply:
okay, I've updated the efforts here.

the best happy-medium I can do (because of mixing `npm` modules and helper scripts) is to write an `npm` script that calls the `run.sh` script, which does all the necessary setup and such. It actually is fairly straight forward and works locally, so it seems like a good pattern to follow.

The names are also fairly obvious (I hope) so it should make it easier to follow along.